kevm-pyk/src/tests/integration/test_conformance.py

```diff
@@ -57,66 +62,97 @@ def _assert_exit_code_zero(pattern: Pattern) -> None:
-    _test(test_file, 'DEFAULT', 'VMTESTS', 1, True)
+def test_vm(gst_data: dict[Path, dict[str, Any]], test_file: Path, test_id: str) -> None:
+    _test(gst_data, TEST_DIR / test_file, test_id, 'DEFAULT', 'VMTESTS', 1, True)
 
 
 @pytest.mark.skip(reason='failing / slow VM tests')
 @pytest.mark.parametrize(
-    'test_file',
+    'test_file, test_id',
     REST_VM_TESTS,
-    ids=[str(test_file.relative_to(VM_TEST_DIR)) for test_file in REST_VM_TESTS],
+    ids=[f'{test_file}:{test_id}' for test_file, test_id in REST_VM_TESTS],
 )
-def test_rest_vm(test_file: Path) -> None:
-    _test(test_file, 'DEFAULT', 'VMTESTS', 1, True)
+def test_rest_vm(gst_data: dict[Path, dict[str, Any]], test_file: Path, test_id: str) -> None:
+    _test(gst_data, TEST_DIR / test_file, test_id, 'DEFAULT', 'VMTESTS', 1, True)
 
 
-BCHAIN_TEST_DIR: Final = TEST_DIR / 'BlockchainTests/GeneralStateTests'
-ALL_BCHAIN_TESTS: Final = tuple(BCHAIN_TEST_DIR.glob('**/*.json'))
-BCHAIN_TESTS: Final = tuple(
-    test_file for test_file in ALL_BCHAIN_TESTS if test_file not in (list(SKIPPED_TESTS) + list(ALL_VM_TESTS))
-)
-REST_BCHAIN_TESTS: Final = tuple(
-    test_file for test_file in ALL_BCHAIN_TESTS if test_file in SKIPPED_TESTS and test_file not in ALL_VM_TESTS
-)
+EXCLUDED_TESTS = SKIPPED_TESTS.union(ALL_VM_TESTS)
+BCHAIN_TESTS: Final = tuple(test for test in ALL_BCHAIN_TESTS if test not in EXCLUDED_TESTS)
+REST_BCHAIN_TESTS: Final = tuple(test for test in ALL_BCHAIN_TESTS if test in EXCLUDED_TESTS)
+
+
+@pytest.fixture(scope='session')
+def gst_data() -> dict[Path, dict[str, Any]]:
+    all_gst_files: Final = {TEST_DIR / file for file, _ in ALL_BCHAIN_TESTS + ALL_VM_TESTS}
+    return {file: json.loads(file.read_text()) for file in all_gst_files}
 
 
 @pytest.mark.parametrize(
-    'test_file',
+    'test_file, test_id',
     BCHAIN_TESTS,
-    ids=[str(test_file.relative_to(TEST_DIR)) for test_file in BCHAIN_TESTS],
+    ids=[f'{test_file}:{test_id}' for test_file, test_id in BCHAIN_TESTS],
 )
-def test_bchain(test_file: Path) -> None:
-    _test(test_file, 'SHANGHAI', 'NORMAL', 1, True)
+def test_bchain(gst_data: dict[Path, dict[str, Any]], test_file: Path, test_id: str) -> None:
+    _test(gst_data, TEST_DIR / test_file, test_id, 'SHANGHAI', 'NORMAL', 1, True)
 
 
 @pytest.mark.skip(reason='failing / slow blockchain tests')
 @pytest.mark.parametrize(
-    'test_file',
+    'test_file, test_id',
     REST_BCHAIN_TESTS,
-    ids=[str(test_file.relative_to(TEST_DIR)) for test_file in REST_BCHAIN_TESTS],
+    ids=[f'{test_file}:{test_id}' for test_file, test_id in REST_BCHAIN_TESTS],
 )
-def test_rest_bchain(test_file: Path) -> None:
-    _test(test_file, 'SHANGHAI', 'NORMAL', 1, True)
+def test_rest_bchain(gst_data: dict[Path, dict[str, Any]], test_file: Path, test_id: str) -> None:
+    _test(gst_data, TEST_DIR / test_file, test_id, 'SHANGHAI', 'NORMAL', 1, True)
+
+
+def create_csv_files(gst_data: dict[Path, dict[str, Any]]) -> None:
```

Comment:
Does this sort the entries?

Reply:
Yes, using `writer.writerows(sorted(content))`.

Reply:
Ah, I missed that. Thanks!